### ISSUE TITLE
[a11y] MediaQuery `accessibleNavigation` unable to detect Android Voice Access 

### ISSUE THREAD

**Issue by liumcse** [flutter/flutter#109521]
Copied from internal bug b/242473326

IIUC, `MediaQuery.of(context).accessibleNavigation` should return true when accessible navigation is turned on. However, this is currently returning false when Android Voice Access is on, making it unable to detect the status of Voice Access.

This should be easily reproducible by checking the value of `MediaQuery.of(context).accessibleNavigation` when Voice Access is turned on.

A minimum reproduce: https://github.com/liumcse/bug_voice_access_repo

**Comment by exaby73** [comment]
Hello @liumcse. Can provide the output of the following:

- `flutter doctor -v`
- 'flutter run -v`

I also want to confirm with you, by voice access, you mean the feature that [this app](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.voiceaccess) provides, right?

**Comment by liumcse** [comment]
```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5 21G72 darwin-arm, locale en-SG)
    • Flutter version 3.0.5 at /Users/xming/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2
```

Would this be sufficient?

Yes, I can confirm that's the app.

**Comment by chunhtai** [comment]
I did some research and found out there is no native API to detect whether switch access or voice access is running. The google internal a11y guide also discourage people from changing the UI based on which assistive technology is running. I will need to reach out to them to see if I can learn more

**Comment by chunhtai** [comment]
asked a question in yaqs/4273359693490946048

**Comment by chunhtai** [comment]
Android team steer toward individual API such as [getRecommendedTimeoutMillis](https://developer.android.com/reference/android/view/accessibility/AccessibilityManager#getRecommendedTimeoutMillis(int,%20int)) for each targeting use case instead of exposing the assistive technology state and let developer guess what UI should behave. I think we may need to rethink our approach.

**Comment by chunhtai** [comment]
finished investigation. There isn't a good way to use getRecommendedTimeoutMillis due to several reasons
1. in native Android view can synchronously call this method to get timeout minute during build time, but Flutter can't call into native API synchronously.
2. We can try to cache this value in the AccessibilityFeature, but there isn't a way to detect when the value may change. For example, you can't tell whether voice access or switch access is turn on or off.

I think the best way to move forward is to apply some workaround to 'guess' whether voice access or switch access is turned on and continue to support the existing accessibleNavigation API. One way to do that will be try to keep track whether android has accessed the info of the semantics tree.
